### ISSUE TITLE
entry.isIntersecting always returns true if observed element or parent has an opacity or transition style

### ISSUE THREAD

**Issue by dverbovyi** [verlok/lazyload#293]
*Are you reporting a bug or requesting a feature?*
Bug

*What is happening?*
If the observed element has CSS **opacility** rule (or its close parent) **entry.isIntersection** always be **true** even for below the fold elements (invisible or out of viewport). Considering this, my team have faced with issue when elements load immediately instead of being postponed.

*What is the expected behaviour?*
The elements which are not in viewport have not to be loaded.

*Which version of LazyLoad are you using?*
10.x

*In which browser(s) and browser version(s) are you experiencing the problem?*
The issue related to Chrome <=71. Here is issue in the [Intersection API](https://github.com/w3c/IntersectionObserver/issues/322) 
and here is official [issue in Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=910741#c13) which was fixed and will available with new Chrome 72 release.

Until Chrome not released new version, I suggest you don't rely on entry.isIntersecting property at all and just check `intersectionRatio > 0`. Also IE doesn't support `isIntersecting`.

All is need to fix is to update 1 line in your code. After that all is works for me perfectly:

export const isIntersecting = entry => 
    ~~entry.isIntersecting ||~~ entry.intersectionRatio > 0;

I've created a PR https://github.com/verlok/lazyload/pull/294

**Comment by verlok** [comment]
True, `entry.isIntersecting` is affected by the bug and it needs fallback because it's `null` on some versions of MS Edge, but `entry.intersectionRatio` alone is not enough because it could be `0` on some intersecting elements.

If it's a Chrome bug and it will be fixed, I can't see the reason to insert a workaround here in the master code. If your case is the one affected by the bug, isn't it better to fork the repo and remove the `isIntersecting` condition from your copy?

**Comment by verlok** [comment]
@dverbovyi I don't remember what is the specific case, I would need to test it again on the most recent versions of MS Edge. Will do.

**Comment by verlok** [comment]
Fixed in version 11.0.2

**Comment by davejamesmiller** [comment]
I just found a case where `entry.intersectionRatio = 0` while `entry.isIntersecting = true`. I'm not sure how to reproduce it, but it's on a large gallery of images and happens to every image on a particular row, causing that row to not load. The images are sized in percentages, so they have a fractional height (231.047px), in case that matters, but there are also a number of absolutely & relatively positioned elements (the layout is generated by a JS plugin) that may also contribute to it being `0`.

I added this `console.log`:

```js
  var isIntersecting = function isIntersecting(entry) {
    console.log(entry.intersectionRatio, entry.isIntersecting);
    return entry.intersectionRatio > 0;
  };
```

And got this output (note: there are 4 images per row):

```
// Initial page load:
0 false
1 true
0.7310053110122681 true
0.7324966788291931 true
0.7332969307899475 true
0.7310053110122681 true
0.730560302734375 true
0.7320507168769836 true
0.7328505516052246 true
0.730560302734375 true
0.7301152944564819 true
0.7316048741340637 true
0.7324041724205017 true
0.7301152944564819 true
0.08727166801691055 true
0.08744971454143524 true
0.08754526078701019 true
0.08727166801691055 true
296 times: 0 false
// On scroll:
4 times: 0.09810027480125427 true
4 times: 0.08227764815092087 true
4 times: 0.012658100575208664 true
4 times: 0.09810027480125427 true
4 times: 0.003164525143802166 true
4 times: 0.009493575431406498 true
4 times: 0.012658100575208664 true
4 times: 0.015822624787688255 true
12 times: 0.006329050287604332 true
4 times: 0 true
4 times: 0.003164525143802166 true
```

I'm not sure what's the best fix generally, but I have switched it to `entry.isIntersecting || entry.intersectionRatio > 0` for now to make it work correctly (in Chrome 72 at least).

**Comment by verlok** [comment]
Thank you for your test @davejamesmiller

I guess now that Chrome 72 is out we can switch it back to `isIntersecting || intersectionRatio > 0` as it was before. 
I also noticed that in some cases the ratio was 0 while the Boolean was true. 
That doesn’t make any sense to me, but I feel safer to use both values.

**Comment by verlok** [comment]
@davejamesmiller would you please do a pull request against my development branch?

**Comment by verlok** [comment]
@davejamesmiller don't bother, I'm on it.

**Comment by verlok** [comment]
Closed, again, with version 11.0.3.

**Comment by davejamesmiller** [comment]
Great, thanks!
